### ISSUE TITLE
[Bug]: Some events don't shows in Free/busy Availability

### ISSUE THREAD

**Issue by section1** [nextcloud/server#47585]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Some user have an event in their calendar but in free/busy availability the slot show as "Free".
I parse the ICS with different validators and passed successfully. But not sure if it's valid and how to reproduce it.
It something related to RDATE.

### Steps to reproduce

1. Not sure how the user created this meet(thunderbird)...but i append the ICS.

### Expected behavior

The expected behavior is that in the freebusy that slot is taken and don't show as Free.

### Installation method

Community Manual installation with Archive

### Nextcloud Server version

28

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.1

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Additional info

i can Fix this if in file ./3rdparty/sabre/vobject/lib/FreeBusyGenerator.php line 389 change this:

```patch
                         $times = [];
 
-                        if ($component->RRULE) {
+                        if ($component->RRULE || $component->RDATE) {
                             try {
                                 $iterator = new EventIterator($object, (string) $component->UID, $this->timeZone);
```

but not sure if is the ideal solution.

**Comment by kesselb** [comment]
cc @ChristophWurst @miaulalala @SebastianKrupinski

**Comment by SebastianKrupinski** [comment]
Yes this would fix the RDATE issue but we would still have a secondary issue. Yes the iterator should be called if there is a RDATE, the problem is that the default iterator only handles one property, so if there is a RRULE and RDATE only the RRULE will be processed due to a oversight in the Sabre DAV library.

That said, I'll do a Sabre PR for this fix.

**Comment by section1** [comment]
thanks @SebastianKrupinski

**Comment by SebastianKrupinski** [comment]
Upstream fix: https://github.com/sabre-io/vobject/pull/691
